sklearn/decomposition/online_lda.py

```diff
@@ -0,0 +1,668 @@
+        http://www.cs.princeton.edu/~mdhoffma/code/onlineldavb.tar
+
+    """
+
+    def __init__(self, n_topics=10, doc_topic_prior=None, topic_word_prior=None,
+                 learning_method='online', learning_decay=.7, learning_offset=1000.,
+                 max_iter=10, batch_size=128, evaluate_every=-1, total_samples=1e6,
+                 normalize_doc=False, perp_tol=1e-1, mean_change_tol=1e-3,
+                 max_doc_update_iter=100, n_jobs=1, verbose=0,
+                 random_state=None):
+        self.n_topics = n_topics
+        if n_topics <= 0:
+            raise ValueError(
+                'Invalid n_topics parameter: %r' %
+                learning_method)
+
+        self.doc_topic_prior = doc_topic_prior
+        if self.doc_topic_prior is None:
+            self.doc_topic_prior_ = 1. / self.n_topics
+        else:
+            self.doc_topic_prior_ = self.doc_topic_prior
+
+        self.topic_word_prior = topic_word_prior
+        if self.topic_word_prior is None:
+            self.topic_word_prior_ = 1. / self.n_topics
+        else:
+            self.topic_word_prior_ = self.topic_word_prior
+
+        self.learning_method = learning_method
+        if learning_method not in ('batch', 'online'):
+            raise ValueError(
+                'Invalid learning_method parameter: %r' %
+                learning_method)
+        self.learning_decay = learning_decay
+        self.learning_offset = learning_offset
+        self.max_iter = max_iter
+        self.batch_size = batch_size
+        self.evaluate_every = evaluate_every
+        self.total_samples = total_samples
+        self.normalize_doc = normalize_doc
+        self.perp_tol = perp_tol
+        self.mean_change_tol = mean_change_tol
+        self.max_doc_update_iter = max_doc_update_iter
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+        self.random_state = random_state
+
+    def _init_latent_vars(self, n_features):
+        """
+        Initialize latent variables.
+        """
+        self.rng = check_random_state(self.random_state)
+        self.n_iter_ = 1
+        self.n_features = n_features
+        init_gamma = 100.
+        init_var = 1. / init_gamma
+
+        # In the literature, this is called `lambda`
+        self.components_ = self.rng.gamma(
+            init_gamma, init_var, (self.n_topics, n_features))
+        # In the literature, this is `E[log(beta)]`
+        self.dirichlet_component_ = _dirichlet_expectation(self.components_)
+        # In the literature, this is `exp(E[log(beta)])`
+        self.exp_dirichlet_component_ = np.exp(self.dirichlet_component_)
+
+    def _e_step(self, X, cal_diff):
+        """
+        E-step
+
+        parameters
+        ----------
+        X : sparse matrix, shape=(n_samples, n_features)
+            Document word matrix.
+
+        cal_diff : boolean
+            parameter that indicate to calculate differene in `component_` or not.
```

Comment:
maybe a more natural shorthand would be ``calc_diff`` or ``calc_change`` here and also in the ``_update_doc_distribution`` function.